modules/tinymce/src/core/main/ts/fmt/FontInfo.ts

```diff
@@ -5,26 +5,54 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
-import { Fun, Obj, Optional } from '@ephox/katamari';
-import { Attribute, Compare, Css, SugarElement, SugarNode, TransformFind } from '@ephox/sugar';
+import { Arr, Obj, Optional, Fun } from '@ephox/katamari';
+import { Attribute, Css, SugarElement, SugarNode } from '@ephox/sugar';
 import DOMUtils from '../api/dom/DOMUtils';
+import * as NodeType from '../dom/NodeType';
+import * as RangeWalk from '../selection/RangeWalk';
+import { isNode } from './FormatUtils';
 
-const legacyPropNames: Record<string, string> = {
+export type FontProp = 'font-size' | 'font-family';
+
+const legacyPropNames: Record<FontProp, string> = {
   'font-size': 'size',
   'font-family': 'face'
 };
 
-const getSpecifiedFontProp = (propName: string, rootElm: Element, elm: HTMLElement): Optional<string> => {
-  const getProperty = (elm: SugarElement) => Css.getRaw(elm, propName).orThunk(() => {
-    if (SugarNode.name(elm) === 'font') {
-      return Obj.get(legacyPropNames, propName).bind((legacyPropName) => Attribute.getOpt(elm, legacyPropName));
-    } else {
-      return Optional.none();
-    }
-  });
-  const isRoot = (elm: SugarElement) => Compare.eq(SugarElement.fromDom(rootElm), elm);
+const collect = (dom: DOMUtils, rngOrNode: Range | Node, rootNode: Element, propName: FontProp): string[] => {
+  const collector: string[] = [];
+
+  const process = (nodes: Node[], limit: Node, getComputed: boolean) => {
+    Arr.each(nodes, (node) => {
+      const matches = dom.getParents(node, '*[style],font', limit);
+      Arr.findMap(matches, (match) => getSpecifiedFontProp(propName, SugarElement.fromDom(match)))
+        .orThunk(() => {
+          // Cannot find computed style on text node so have get parent node if required
+          const elm = NodeType.isText(node) ? dom.getParent(node, '*', rootNode) : node as Element;
+          return getComputed ? getComputedFontProp(propName, elm) : Optional.none();
+        })
+        .each((value) => {
+          // Make sure value is unique
+          if (Arr.forall(collector, (existing) => existing !== value)) {
+            collector.push(value);
+          };
+        });
+
+      // Don't need to worry about potentially getting the computed value for children as if no fonts are found on the child,
+      // it will inherit the 'node' font value which may be computed
+      process(Arr.from(node.childNodes), node, false);
+    });
+  };
+
+  if (isNode(rngOrNode)) {
```

Comment:
Hmm... this will cause very different behaviours, as `RangeWalk` walks from the bottom up. It's also going to cause it to walk from the bottom, up one level and then walk back down since `process` recursively checks child nodes.

That doesn't seem like a very good solution for collecting the fonts here. I'll have to think of other options, but the first option that comes to mind is when a node is passed, create a range around that node and remove the recursive calls in `process`.